presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1094,6 +1096,18 @@ std::vector<std::string> PrestoServer::registerConnectors(
       // make sure connector type is supported
       getPrestoToVeloxConnector(connectorName);
 
+      // Validate hive configs.
+      if (connectorName == kHive || connectorName == kHiveHadoop2) {
```

Comment:
Why is the validation done in this manner ? It could be part of the HivePrestoToVeloxConnector constructor itself.